notebooks/GWAS_blood_pressure_p2/main.py

```diff
@@ -270,390 +254,17 @@
-    p = hl.plot.scatter(pcs.scores[0],
-                        pcs.scores[1],
-                        xlabel='PC1', ylabel='PC2')
-    bokeh_io.show(p)
-with herzog.Cell("markdown"):
+    # Save sample metadata and update data table
+    Now that we've explored the phenotypes, the next step is to save and push our results back to the workspace data model. Once we populate the data model, we can use it for downstream analyses.
     """
-    ### Decide whether genetic stratification should be included in your association tests
 
-    If your project shows distinct clusters of samples in your PCA, you need to account for this genetic stratification by including the first principal components as covariates in your analysis. This can be done using the annotate_cols function.
-    """
-with herzog.Cell("python"):
-    # Use describe to find the PC fields you want to keep
-    pcs.describe()
-with herzog.Cell("python"):
-    # Add to the matrix table
-    mt = mt.annotate_cols(scores=pcs[mt.s].scores)
-with herzog.Cell("markdown"):
-    """
-    # Generate a genetic relatedness matrix (GRM)
-    Hail has built-in functions for generating a GRM. A GRM can also account for population stratification and cryptic genetic relatedness in our cohort.
-    """
-# On chr 1 from Amish this only took 46 seconds but on all Amish chromosomes it seemed to have hung
-with herzog.Cell("python"):
-    # Calculate the GRM
-    # WARNING: This can take a very long time to complete!
-    start_grm_time = time.time()
 
-    grm = hl.genetic_relatedness_matrix(mt.GT).to_numpy()
 
-    elapsed_grm_time = time.time() - start_grm_time
-    print(timedelta(seconds=elapsed_grm_time))
-with herzog.Cell("python"):
-    # Get the right sample order
-    ind_order = mt.s.collect()
-with herzog.Cell("python"):
-    # GRM to data frame
-    rel_pd = pd.DataFrame(data=grm,
-                          index=ind_order,
-                          columns=ind_order)
-with herzog.Cell("python"):
-    # Export GRM
-    rel_pd.to_csv(kinship_out)
-with herzog.Cell("markdown"):
-    """
-    # Save sample metadata and update data table
-    Now that we've explored the phenotypes and generated relatedness measures, the next step is to save and push our results back to the workspace data model. Once we populate the data model, we can use it for downstream analyses.
-
-    ## Convert the phenotype data to the correct format
-    """
-with herzog.Cell("python"):
-    # Gather the sample metadata
-    samples_traits_for_analysis = (
-        mt.cols()
-    )
-with herzog.Cell("python"):
-    # Select only the columns we want to keep
-    samples_traits_for_analysis = (
-        samples_traits_for_analysis
-        .key_by()
-        .select('s', 'pheno')
-    )
-with herzog.Cell("python"):
-    # Convert the Hail matrix back to a pandas dataframe
-    samples_traits_for_analysis = (
-        samples_traits_for_analysis
-        .to_pandas()
-    )
 with herzog.Cell("python"):
```

Comment:
I don't think you need the rename step below in the phenotype notebook. Only a Hail matrix renames the column headers to have "pheno.___"

Reply:
You also can remove the extra head function. Only keep the to.csv()